acme/client.go

```diff
@@ -100,6 +100,24 @@ func NewClient(caDirURL string, user User, keyBits int) (*Client, error) {
 	return &Client{directory: dir, user: user, jws: jws, keyBits: keyBits, solvers: solvers}, nil
 }
 
+// SetChallengeProvider specifies a custom provider that will make the solution available
+func (c *Client) SetChallengeProvider(challenge string, p ChallengeProvider) {
+	switch challenge {
+	case "http-01":
+		if chlng, ok := c.solvers[challenge]; ok {
+			chlng.(*httpChallenge).provider = p
+		}
+	case "tls-sni-01":
+		if chlng, ok := c.solvers[challenge]; ok {
+			chlng.(*tlsSNIChallenge).provider = p
+		}
+	case "dns-01":
+		if chlng, ok := c.solvers[challenge]; ok {
```

Comment:
It does not apply for the default `NewClient` you are right. But I want to add a `New...` function which does not add any default solvers but lets the user configure it completely to his liking. For this case it would be great to have the functionality already there.